[16.0][OU-ADD] website_slides_survey: Migration to 16.0 (#4980)
OCA/OpenUpgrade: openupgrade_scripts/scripts/website_slides_survey/16.0.1.0/upgrade_analysis_work.txt

---Models in module 'website_slides_survey'---
---Fields in module 'website_slides_survey'---
website_slides_survey / slide.slide              / name (False)                  : NEW mode: modify, hasdefault: compute
# NOTHING TO DO: The field exists in the website_slides module as storable; just changed it to computed.

website_slides_survey / slide.slide              / slide_category (False)        : NEW selection_keys: ['article', 'certification', 'document', 'infographic', 'quiz', 'video'], mode: modify

> # DONE pre-migration: The field exists in the website_slides module, the migration script is done on website_slides

Reviewer: ```suggestion
# NOTHING TO DO: The field exists in the website_slides module, the migration script is done on website_slides
```

Author: Done, thanks

website_slides_survey / slide.slide              / slide_type (False)            : selection_keys is now '['article', 'certification', 'doc', 'google_drive_video', 'image', 'pdf', 'quiz', 'sheet', 'slides', 'vimeo_video', 'youtube_video']' ('['certification', 'document', 'infographic', 'presentation', 'quiz', 'video', 'webpage']')

> # Done: Fill the correct value when slide_category=certification

Reviewer 2: ```suggestion
# DONE: pre-migration: Filled with "certification" value when slide_category=certification
```

---XML records in module 'website_slides_survey'---
DEL ir.actions.act_window: website_slides_survey.survey_survey_action_slides_report
DEL ir.ui.menu: website_slides_survey.website_slides_menu_report_certification

> # NOTHING TO DO: Handle by ORM

Reviewer 2: ```suggestion
# NOTHING TO DO: Handled by ORM
```

NEW ir.actions.act_window.view: website_slides_survey.survey_survey_action_slides_view_form
NEW ir.actions.act_window.view: website_slides_survey.survey_survey_action_slides_view_kanban
NEW ir.actions.act_window.view: website_slides_survey.survey_survey_action_slides_view_tree

> # NOTHING TO DO: New features

Reviewer 2: ```suggestion
# NOTHING TO DO: it's noupdate=0 stuff
```

ir.model.constraint: website_slides_survey.constraint_slide_slide_check_certification_preview (changed definition: is now 'check(slide_category != 'certification' or is_preview = false)' ('check(slide_type != 'certification' or is_preview = false)'))
ir.model.constraint: website_slides_survey.constraint_slide_slide_check_survey_id (changed definition: is now 'check(slide_category != 'certification' or survey_id is not null)' ('check(slide_type != 'certification' or survey_id is not null)'))

> # DONE pre-migration: safely delete constraint to recreate it

Reviewer 2: ```suggestion
# DONE: pre-migration: safely delete constraints for letting ORM to recreate them
```

NEW ir.ui.view: website_slides_survey.res_config_settings_view_form
NEW ir.ui.view: website_slides_survey.slide_sidebar_done_button
NEW ir.ui.view: website_slides_survey.slide_slide_partner_view_form
NEW ir.ui.view: website_slides_survey.slide_slide_partner_view_search
NEW ir.ui.view: website_slides_survey.slide_slide_partner_view_tree
NEW ir.ui.view: website_slides_survey.survey_survey_view_kanban
NEW ir.ui.view: website_slides_survey.survey_survey_view_search_slides
NEW ir.ui.view: website_slides_survey.survey_survey_view_tree_slides

> # NOTHING TO DO: New features

Reviewer 2: This can be grouped with the previous ir noupdate=0 stuff.
